src/main/java/ogc/rs/database/DatabaseServiceImpl.java

```diff
@@ -161,8 +161,8 @@ public Future<JsonObject> getFeatures(String collectionId, Map<String, String> q
                 Promise<Void> bboxPromise = Promise.promise();
 
                 if (queryBbox != null && tokenBbox != null) {
-                    // Use PostGIS to check bbox intersection
-                    String sql = "SELECT ST_Intersects(" +
+                    // Use PostGIS to check if query bbox is within token bbox using ST_Within
+                    String sql = "SELECT ST_Within(" +
```

Comment:
I think that when it comes to the `bbox` query param check, it makes sense to stick to using `Intersects`. e.g. 
- if I have a token bbox across all South Indian states, and 
- I make a query with bbox just including Karnataka and Kerala

the query bbox may not necessarily be contained by the token bbox, it will surely intersect. Just because it's not contained by it, it does not mean that there are no features common to both bbox's